proxy.c

```diff
@@ -29,6 +29,13 @@
     #include <arpa/inet.h>
     #include <netinet/tcp.h>
     #include <netdb.h>
+
+    #ifdef __linux__
+        // Copied from musl's socket.h to not use _GNU_SOURCE
+        #ifdef __GLIBC__
+        extern int accept4(int, struct sockaddr *__restrict, socklen_t *__restrict, int);
```

Comment:
why avoid _GNU_SOURCE? this is a crutch solution.

Reply:
GNU defines its functions which overlap with POSIX functions, but they sometimes behave differently. So even if the project probably still not affected by such problem i think its better to not define _GNU_SOURCE for big files. Defining _GNU_SOURCE only on linux is also bad idea IMO.

Musl libc already defines accept4 without any _GNU_SOURCE:
https://git.musl-libc.org/cgit/musl/tree/include/sys/socket.h#n389

However glibc at the same requires _GNU_SOURCE:
https://sourceware.org/git/?p=glibc.git;a=blob;f=include/sys/socket.h#l95
https://sourceware.org/git/?p=glibc.git;a=blob;f=include/sys/socket.h#l159

So if there a better way to define accept4 without _GNU_SOURCE i would be glad to know it.

Reply:
I can only think that probably using directly syscall instead libc function for accept4 will be better solution because Linux tries to keep syscalls stable.

Reply:
```
#ifdef _WIN32
    ...
#else
    #define _GNU_SOURCE
    #include <sys/socket.h>
    #undef _GNU_SOURCE
    #include <unistd.h>
    ...
#endif

#include <stdint.h>
#include <stdio.h>
```